cmd/exchange_template/exchange_template.go

```diff
@@ -130,112 +152,99 @@ func main() {
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tReadme.Execute(r1, exch)
-	r1.Close()
+	for x := range outputFiles {
+		var tmpl *template.Template
+		tmpl, err = template.New(outputFiles[x].Name).ParseFiles(outputFiles[x].TemplateFile)
+		if err != nil {
+			return fmt.Errorf("%s template error: %s", outputFiles[x].Name, err)
+		}
 
-	tMain, err := template.New("main").ParseFiles("main_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeMain)
-	m1, err := os.OpenFile(exchangeMain, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tMain.Execute(m1, exch)
-	m1.Close()
+		filename := outputFiles[x].Filename
+		if outputFiles[x].FilePostfix != "" {
+			filename = exch.Name + outputFiles[x].FilePostfix
+		}
 
-	tTest, err := template.New("test").ParseFiles("test_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeTest)
-	t1, err := os.OpenFile(exchangeTest, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tTest.Execute(t1, exch)
-	t1.Close()
+		outputFile := filepath.Join(exchangeDirectory, filename)
+		newFile(outputFile)
+		var f *os.File
+		f, err = os.OpenFile(outputFile, os.O_WRONLY, 0770)
+		if err != nil {
+			return err
+		}
 
-	tType, err := template.New("type").ParseFiles("type_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeTypes)
-	ty1, err := os.OpenFile(exchangeTypes, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
+		if err = tmpl.Execute(f, exch); err != nil {
+			f.Close()
+			return err
+		}
+		f.Close()
 	}
-	tType.Execute(ty1, exch)
-	ty1.Close()
 
-	tWrapper, err := template.New("wrapper").ParseFiles("wrapper_file.tmpl")
+	cmd := exec.Command("go", "fmt")
+	cmd.Dir = exchangeDirectory
+	out, err := cmd.Output()
 	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
+		return fmt.Errorf("unable to go fmt. output: %s err: %s", out, err)
 	}
-	newFile(exchangeWrapper)
-	w1, err := os.OpenFile(exchangeWrapper, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tWrapper.Execute(w1, exch)
-	w1.Close()
 
-	err = exec.Command("go", "fmt", exchangeDirectory).Run()
+	configTestFile.Exchanges = append(configTestFile.Exchanges, newExchConfig)
+	err = configTestFile.SaveConfig(exchangeConfigPath, false)
 	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool go fmt error ", err)
+		return err
 	}
 
-	err = exec.Command("go", "test", exchangeDirectory).Run()
+	cmd = exec.Command("go", "test")
```

Comment:
Here's a fun nit, using this application, you cannot create an exchange that starts with `s` as variable names collide:
```

// SubmitOrder submits a new order
func (s *superlol3) SubmitOrder(s *order.Submit) (order.SubmitResponse, error) {
	var submitOrderResponse order.SubmitResponse
	if err := s.Validate(); err != nil {
		return submitOrderResponse, err
	}
	return submitOrderResponse, common.ErrNotYetImplemented
}
```

I would suggest that the template cannot have any single char variables as params as they have the potential to collide with any exchange owned function in `(c *coolExchangeFourTwoZero)`

I select this line because this is where it will fail right now when you run go test creating an exchange starting with `s`

Reply:
I was actually thinking about this and then the thought was free'd, I'll push a fix which just makes it the first two characters to avoid collisions

Reply:
Fix pushed!